### PR TITLE
chore:Refactor Repository Component to use hooks

### PR DESCRIPTION
**Pull request by bpedersen**
**Type:**

The following has been addressed in the PR:

*  There is a related issue?
*  Unit or Functional tests are included in the PR

**Description:**

### CODE REVIEW

File: src/components/Repository/Repository.tsx [hunk 1]
```diff
@@ -11,47 +10,36 @@ import List from '../../muiComponents/List';
...
+
+		if (!url || isURL(url) === false) {
+			return null;
+		}
+
+		return (
+			<Fragment>
+				<List dense={true} subheader={<StyledText variant="subtitle1">{'Repository'}</StyledText>}>
+					<RepositoryListItem button={true}>
+						<Avatar src={git} />
```
Comment (ayusharma) [inline src/components/Repository/Repository.tsx]: I would recommend renaming `git` to a meaningful name. eg. `gitImage`.

File: src/components/Repository/Repository.test.tsx [hunk 2]
```diff
@@ -67,7 +67,7 @@ describe('<Repository /> component', () => {
 
     mockPackageMeta.mockImplementation(() => packageMeta);
 
-    const wrapper = mount(<Repository />);
-    expect(wrapper.html()).toEqual('');
+    const { container } = render(<Repository />);
+    expect(container.innerHTML).toEqual('');
```
Comment (priscilawebdev) [inline src/components/Repository/Repository.test.tsx]:
```suggestion
    expect(container.firstChild).toEqual('');
```

File: src/components/Repository/Repository.test.tsx [hunk 3]
```diff
@@ -51,8 +51,8 @@ describe('<Repository /> component', () => {
...
```
Comment (priscilawebdev) [inline src/components/Repository/Repository.test.tsx]:
```suggestion
    expect(container.firstChild).toEqual('');
```

File: src/components/Repository/Repository.tsx [hunk 4]
```diff
@@ -11,47 +10,36 @@ import List from '../../muiComponents/List';
...
+						<RepositoryListItemText primary={renderContent(url)} />
+					</RepositoryListItem>
+				</List>
+			</Fragment>
+		);
+	};
+	return <DetailContextConsumer>
```
Comment (priscilawebdev) [inline src/components/Repository/Repository.tsx]: Please use the hook `useContext`

File: src/components/Repository/Repository.tsx [hunk 5]
```diff
@@ -11,47 +10,36 @@ import List from '../../muiComponents/List';
...
-  }
-}
+const Repository: React.FC = props => {
+	const renderRepositoryText = (url: string) => (
+		<GithubLink href={url} target="_blank">
+			{url}
+		</GithubLink>
+	);
+
+	const renderContent = (url: string) => <CopyToClipBoard text={url}>{renderRepositoryText(url)}</CopyToClipBoard>;
```
Comment (priscilawebdev) [inline src/components/Repository/Repository.tsx]: IMO you could ignore by adding `/* eslint-disable react/jsx-max-depth */` it doesn't make sense to keep such a little piece of code separate like this

File: src/components/Repository/Repository.test.tsx [hunk 6]
```diff
@@ -40,8 +40,8 @@ describe('<Repository /> component', () => {
 
     mockPackageMeta.mockImplementation(() => packageMeta);
 
-    const wrapper = mount(<Repository />);
-    expect(wrapper.html()).toMatchSnapshot();
+    const { container } = render(<Repository />);
+    expect(container.innerHTML).toMatchSnapshot();
```
Comment (priscilawebdev) [inline src/components/Repository/Repository.test.tsx]:
```suggestion
    expect(container.firstChild).toMatchSnapshot();
```

File: src/components/Repository/Repository.tsx [hunk 7]
```diff
@@ -11,47 +10,36 @@ import List from '../../muiComponents/List';
...
-      </Fragment>
-    );
-  };
-
-  private renderContent(url: string): ReactElement<HTMLElement> {
-    return <CopyToClipBoard text={url}>{this.renderRepositoryText(url)}</CopyToClipBoard>;
-  }
-}
+const Repository: React.FC = props => {
+	const renderRepositoryText = (url: string) => (
```
Comment (priscilawebdev) [inline src/components/Repository/Repository.tsx]: I would keep this little piece of code together

File: src/components/Repository/Repository.tsx [hunk 8]
```diff
@@ -11,47 +10,36 @@ import List from '../../muiComponents/List';
...
+
+	const renderRepository = (packageMeta: PackageMetaInterface) => {
+		const { repository: { url = null } = {} } = packageMeta.latest;
+
+		if (!url || isURL(url) === false) {
+			return null;
+		}
+
+		return (
+			<Fragment>
```
Comment (priscilawebdev) [inline src/components/Repository/Repository.tsx]:
```suggestion
			<>
```

File: src/components/Repository/Repository.tsx [hunk 9]
```diff
@@ -11,47 +10,36 @@ import List from '../../muiComponents/List';
...
+				<List dense={true} subheader={<StyledText variant="subtitle1">{'Repository'}</StyledText>}>
+					<RepositoryListItem button={true}>
+						<Avatar src={git} />
+						<RepositoryListItemText primary={renderContent(url)} />
+					</RepositoryListItem>
+				</List>
+			</Fragment>
```
Comment (priscilawebdev) [inline src/components/Repository/Repository.tsx]:
```suggestion
			</>
```

**Comment by priscilawebdev** [comment]
@bpedersen thank you for contributing to Verdaccio 🤩Could you please review my comments? I suggested some minor changes. I believe that we could keep the whole code together... I mean this is not a large component..there is no need to split the code into little pieces...we might change that lint rule 😉

File: src/components/Repository/Repository.test.tsx (hunk 2 above)
Comment (juanpicado) [reply src/components/Repository/Repository.test.tsx]: Why @priscilawebdev ?

File: src/components/Repository/Repository.tsx (hunk 4 above)
Comment (juanpicado) [reply src/components/Repository/Repository.tsx]: https://reactjs.org/docs/hooks-reference.html#usecontext

`useContext` is really simple, let us know whether you need support here.

File: src/components/Repository/Repository.tsx (hunk 5 above)
Comment (juanpicado) [reply src/components/Repository/Repository.tsx]: Agree, in this case makes totally sense what @priscilawebdev suggest.

**Comment by priscilawebdev** [comment]
Hi @bpedersen. 

I hope you don't mind if I take this over :) I would like to finish this refactor  (class to components) as soon as possible, so then we can concentrate on the theme.

Thank you!! 🙏

**Comment by priscilawebdev** [comment]
I have done this refactor here: https://github.com/verdaccio/ui/pull/323 for this reason I am closing this PR 😉

**Pull request closed without merging**